github_rules/github_repo_initial_access.py

```diff
@@ -0,0 +1,23 @@
+from panther_oss_helpers import get_string_set, put_string_set
+
+
+def rule(event):
+    if event.get("action").startswith("git.") and not event.get("repository_public", False):
+        # trigger on any of the git actions, http or ssh
+        key = get_key(event)
+        previous_access = get_string_set(key)
+        if not previous_access:
+            put_string_set(key, key)
+            return True
+    return False
+
+
+def title(event):
+    return (
+        f"A user [{event.udm('actor_user')}] accessed a private repository "
+        f"[{event.get('repo', '<UNKNOWN_REPO>')}] for the first time."
+    )
+
+
+def get_key(event):
+    return str(event.udm("actor_user")) + ":" + str(event.get("repo"))
```

Comment:
Let's also prepend the rule ID (either hardcoded or by using the `__name__ ` variable) so that if future detections wish to use the KV table and key off the `actor_user` and `repo` fields there's no conflict.